src/headless_browser.ts

```diff
@@ -228,12 +218,14 @@ export class HeadlessBrowser {
     const command = `document.body.innerText.indexOf('${text}') >= 0`;
     const res = await this.sendWebSocketMessage("Runtime.evaluate", {
       expression: command,
-    });
-    this.checkForErrorResult((res as DOMOutput), command);
-    // Tried and tested, and `result` is `{result: { type: "boolean", value: false } }`
-    const exists = ((res as DOMOutput).result as SuccessResult)
-      .value as boolean;
-    if (exists !== true) {
+    }) as { // Tried and tested
+      result: {
```

Comment:
can we make this a `Result` type? not sure if you tried that already

Reply:
The thing is, the output from the events are all different and vary a lot which is why I went down the route of specifying the type per method - at first I did try make an interface for every event, but I decided against that and I’m trying to remember why

Reply:
just seeing if we could DRY up the code. no biggie

Reply:
Wait did you mean create a new interface called Result which this method will use (and any others that are casting to the same type)?

Reply:
Un-resolving because I think you make a good point

Reply:
Actually, going to re-resolve this as the output from all the messages can vary so much, that it's hard to say if any others will return the same amount of data, so `Result` could be misleading